cmd/app/list_discussions.go

```diff
@@ -30,20 +39,30 @@ type DiscussionsResponse struct {
 	Emojis              map[int][]*gitlab.AwardEmoji `json:"emojis"`
 }
 
-type SortableDiscussions []*gitlab.Discussion
+type SortableDiscussions struct {
+	Discussions []*gitlab.Discussion
+	SortBy      SortBy
+}
 
-func (n SortableDiscussions) Len() int {
-	return len(n)
+func (d SortableDiscussions) Len() int {
+	return len(d.Discussions)
 }
 
-func (d SortableDiscussions) Less(i int, j int) bool {
-	iTime := d[i].Notes[len(d[i].Notes)-1].CreatedAt
-	jTime := d[j].Notes[len(d[j].Notes)-1].CreatedAt
-	return iTime.After(*jTime)
+func (d SortableDiscussions) Less(i, j int) bool {
+	var iTime, jTime *time.Time
+	if d.SortBy == SortByOriginalComment {
+		iTime = d.Discussions[i].Notes[0].CreatedAt
+		jTime = d.Discussions[j].Notes[0].CreatedAt
+		return iTime.Before(*jTime)
+	} else { // SortByLatestReply
+		iTime = d.Discussions[i].Notes[len(d.Discussions[i].Notes)-1].CreatedAt
+		jTime = d.Discussions[j].Notes[len(d.Discussions[j].Notes)-1].CreatedAt
+		return iTime.After(*jTime)
```

Comment:
Small nitpick, you don't need multiple returns here, you can just return outside of this if/else